lib/fog/core/services_mixin.rb

```diff
@@ -24,16 +24,22 @@ def providers
 
     private
 
+    # This method should be removed once all providers are extracted.
+    # Bundler will correctly require all dependencies automatically and thus
+    # fog-core wont need to know any specifics providers. Each provider will
+    # have to load its dependencies.
     def require_service_provider_library(service, provider)
       require "fog/#{provider}/#{service}"
     rescue LoadError  # Try to require the service provider in an alternate location
+      Fog::Logger.deprecation('Prefer to define all Services within the Provider Namespace instead!')
       require "fog/#{service}/#{provider}"
     end
 
     def service_provider_constant(service_name, provider_name)
-      Fog.const_get(service_name).const_get(*const_get_args(provider_name))
-    rescue NameError  # Try to find the constant from in an alternate location
       Fog.const_get(provider_name).const_get(*const_get_args(service_name))
+    rescue NameError  # Try to find the constant from in an alternate location
+      Fog::Logger.deprecation('Prefer to define all Services within the Provider Namespace instead!')
```

Comment:
Similarly, a more specific error here would probably be less confusing, though I'm much less sure how to word it helpfully here. What do you think?